simulariumio/data_objects/model_meta_data.py

```diff
@@ -0,0 +1,143 @@
+        self.source_code_license_url = (
+            source_code_license_url if source_code_license_url is not None else ""
+        )
+        self.input_data_url = input_data_url if input_data_url is not None else ""
+        self.raw_output_data_url = (
+            raw_output_data_url if raw_output_data_url is not None else ""
+        )
+
+    @classmethod
+    def from_buffer_data(cls, buffer_data: Dict[str, Any]):
+        """
+        Create ModelMetaData from a simularium JSON dict containing buffers
+        """
+        model_info = (
+            buffer_data["trajectoryInfo"]["modelInfo"]
+            if "modelInfo" in buffer_data["trajectoryInfo"]
+            else None
+        )
+        if model_info is None:
+            return None
+        return cls(
+            title=model_info["title"] if "title" in model_info else "",
+            version=model_info["version"] if "version" in model_info else "",
+            authors=model_info["authors"] if "authors" in model_info else "",
+            description=model_info["description"]
+            if "description" in model_info
+            else "",
+            doi=model_info["doi"] if "doi" in model_info else "",
+            source_code_url=model_info["sourceCodeUrl"]
+            if "sourceCodeUrl" in model_info
+            else "",
+            source_code_license_url=model_info["sourceCodeLicenseUrl"]
+            if "sourceCodeLicenseUrl" in model_info
+            else "",
+            input_data_url=model_info["inputDataUrl"]
+            if "inputDataUrl" in model_info
+            else "",
+            raw_output_data_url=model_info["rawOutputDataUrl"]
+            if "rawOutputDataUrl" in model_info
+            else "",
+        )
+
+    def is_default(self):
+        """
+        Check if this ModelMetaData is only holding default data
+        """
+        return (
+            not self.title
+            and not self.version
+            and not self.authors
+            and not self.description
+            and not self.doi
+            and not self.source_code_url
+            and not self.source_code_license_url
+            and not self.input_data_url
+            and not self.raw_output_data_url
+        )
+
+    def __iter__(self):
+        if self.title:
+            yield "title", self.title
+        if self.version:
+            yield "version", self.version
+        if self.authors:
+            yield "authors", self.authors
+        if self.description:
+            yield "description", self.description
+        if self.doi:
+            yield "doi", self.doi
+        if self.source_code_url:
+            yield "sourceCodeUrl", self.source_code_url
+        if self.source_code_license_url:
+            yield "sourceCodeLicenseUrl", self.source_code_license_url
+        if self.input_data_url:
+            yield "inputDataUrl", self.input_data_url
+        if self.raw_output_data_url:
+            yield "rawOutputDataUrl", self.raw_output_data_url
```

Comment:
I looked up some basics of what `__iter__` does but I still don't understand, can you explain for me what this method enables?

Reply:
yeah this let's us do `dict(input_data.meta_data.model_meta_data)` to turn the object into a dict to be saved in json during `TrajectoryConverter._read_trajectory_data()`

Reply:
ooo!